File: Number-of-Pairs-of-Interchangeable-Rectangles.java
class Solution {
    public long interchangeableRectangles(int[][] rectangles) {
        Map<Double, Long> map = new HashMap<>();
        long count = 0;

        for (int[] rect : rectangles) {
            double ratio = (double) rect[0] / rect[1];
            count += map.getOrDefault(ratio, 0L);
            map.put(ratio, map.getOrDefault(ratio, 0L) + 1);
        }

        return count;
    }
}

File: Number-of-Pairs-of-Interchangeable-Rectangles.cpp
class Solution {
public:
    long long interchangeableRectangles(vector<vector<int>>& r) {
        long long n = r.size();
        long long c = 0;
        unordered_map<double, int> mp;

        for (int i = 0; i < n; i++) {
            int x = r[i][0];
            int y = r[i][1];
            double z = (double)x / y;  
            c += mp[z];
            mp[z]++;
        }

        return c;
    }
};